pkg/core/resources/apis/mesh/circuit_breaker_validator.go

```diff
@@ -14,6 +14,13 @@ func (c *CircuitBreakerResource) HasDetectors() bool {
 		c.Spec.Conf.GetDetectors().GetFailure() != nil
 }
 
+func (c *CircuitBreakerResource) HasThresholds() bool {
+	return c.Spec.Conf.GetThresholds().GetMaxConnections() != nil ||
+		c.Spec.Conf.GetThresholds().GetMaxPendingRequests() != nil ||
+		c.Spec.Conf.GetThresholds().GetMaxRetries() != nil ||
+		c.Spec.Conf.GetThresholds().GetMaxRequests() != nil
```

Comment:
According to: https://github.com/kumahq/kuma/blob/ff2e4f2eab0d5da3f08397692a4c69c2b9a9513d/api/mesh/v1alpha1/circuit_breaker.pb.go#L182-L187

`c.Spec.Conf.GetThresholds()` can return nil

```suggestion
	tresholds := c.Spec.Conf.GetThresholds()

	return tresholds != nil ||
		tresholds.GetMaxConnections() != nil ||
		tresholds.GetMaxPendingRequests() != nil ||
		tresholds.GetMaxRetries() != nil ||
		tresholds.GetMaxRequests() != nil
```

Reply:
It would be also good to have test for this I mentioned above

Reply:
That's true, but `GetMaxConnections` will return `nil` if `thresholds` is `nil`. Same for all other autogenerated proto methods. I think we can rely on it and have fewer nil-checks when it comes to protubuf.

Reply:
You are right. I realized it's checking for `x` and not `x.Thresholds`.